File: src/main/kotlin/eu/thelastdodo/tarock20/domain/GameDto.kt
package eu.thelastdodo.tarock20.domain

import eu.thelastdodo.tarock20.entity.enums.GamePhase

data class GameDto(
    val id: String,
    var phase: GamePhase,
    val player: PlayerDto,
    val otherPlayers: List<OtherPlayerDto>,
    var activePlayer: String,
    var foreHand: String,
)

File: tarock20-frontend/src/store/GameDto.ts
import OtherPlayerDto from "./OtherPlayerDto"
import PlayerDto from "./PlayerDto"

export default class GameDto {
    public id: string = ""
    public player: PlayerDto | null = null
    public otherPlayers: OtherPlayerDto[] = []

    public isInGame(playerId: string): boolean {
        return !!this.otherPlayers.find((player) => player.id === playerId) || this.player?.id === playerId
    }

    public static fromJson(json: string): GameDto {
        return Object.assign(new GameDto(), JSON.parse(json))
    }
}